### ISSUE TITLE
Yolov10x - Model card

### ISSUE THREAD

**Issue by vigneshkeerthivasanx** [tenstorrent/tt-metal#17935]
**Brief Exec summary of Yolov10x**

- Perf PR: https://github.com/tenstorrent/tt-metal/pull/20410
- Demo PR: https://github.com/tenstorrent/tt-metal/pull/20255
- Trace PR: https://github.com/tenstorrent/tt-metal/pull/21283
- Supported Resolution: 640x640
- Supported Batch size: 1
- Graph: [yolov10x_640.gv.pdf](https://github.com/user-attachments/files/18842172/yolov10x_640.gv.pdf)
- Reference [Link](https://github.com/ultralytics/ultralytics/tree/main)
- Branch: [vignesh/ttnn_yolov10x](https://github.com/tenstorrent/tt-metal/compare/main...vignesh/ttnn_yolov10x)
- Unit test: [vignesh/ttnn_yolov10x_op_unit_test](https://github.com/tenstorrent/tt-metal/compare/vignesh/ttnn_yolov10x_op_unit_test)
- PCC: 0.999
- FPS: 42
- Deno results : [Link](https://docs.google.com/document/d/1nyCuydgErgt0f9gYnDuANSYX-mwLCOtYZ-MkYU1sBbQ/edit?usp=sharing)

Modules:

1. Bottleneck
2. C2f
3. CIB
4. SCDown
5. C2fCIB
6. SPPF
7. Attention
8. PSA
9. DFL
10. v10Detect

Ops:

1. Conv2d
2. BatchNorm2d
3. SiLU
4. Concat
5. Upsample
6. Softmax
7. sigmoid

Performance Issue:
#21281
Depthwise conv - [#25523](https://github.com/tenstorrent/tt-metal/issues/25523)

**Comment by ssinghalTT** [comment]
Can we put the model branch here? @vigneshkeerthivasanx

**Comment by saichandax** [comment]
Model is in main. https://github.com/tenstorrent/tt-metal/tree/main/models/demos/yolov10x
